### PR TITLE
 calculate average task progress in course

### PR DESCRIPTION
**Pull request by ahmedehab0**
### new Routes:
GET /courses/:courseId/tasks/progress
Purpose: Calculate average task progress for a course.

### new models:
UserTaskProgress

id (integer, Primary Key)
user_id (Foreign Key, References User)
task_id (Foreign Key, References Task)
is_finished (Boolean, default: false)
completed_at (Timestamp, nullable)

### CODE REVIEW

**Comment by ahmedehab0** [comment]
i'm sorry for the changed variable at .env file, I didn't notice that someone has changed the .gitignore file

File: Server/Controllers/taskProgressController.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    const tasks_completed_count = await UserTaskProgress.count({
+        where: {
+            is_finished: true,
+            userId,
+        },
+    });
+
+    const average_progress = (tasks_completed_count / tasks_count) * 100;
+
+    res.status(200).json({average_progress})
```
Comment (AhmedNagehAbdelrazek) [inline Server/Controllers/taskProgressController.js:38]: Why don't you merge the `tasks_count` and `average_progress` together in one response
and it is better to git the whole data to the front to deal with it too
so the res should look like that 
```
{
    "average_progress": 0.5,
    "tasks_count": 10,
    "tasks_completed_count": 5
}
```

File: Server/Routes/courseRoutes.js [hunk 2]
```diff
@@ -13,7 +14,9 @@ router.post("/",protect, verifyRole("educator"),courseValidator, createCourse);
 router.put("/:id",protect, verifyRole("educator"), updateCourse);
 router.delete("/:id",protect, verifyRole("educator"),deleteCourseById);
 
-//course progress Routes
-router.use("/",progressRoutes);
+//progress Routes
+router.use("/", progressRoutes);
+router.use("/", taskProgressRoutes);
```
Comment (AhmedNagehAbdelrazek) [inline Server/Routes/courseRoutes.js:19]: pay attention to how i did that here with the route

**Review by AhmedNagehAbdelrazek — CHANGES_REQUESTED** [review]

File: Server/Controllers/taskProgressController.js (hunk 1 above)
Comment (ahmedehab0) [reply Server/Controllers/taskProgressController.js]: Done

**Review by AhmedNagehAbdelrazek — APPROVED** [review]

**Pull request merged**